src/main/java/com/aws/greengrass/deployment/activator/KernelUpdateActivator.java

```diff
@@ -61,6 +67,23 @@ public void activate(Map<String, Object> newConfig, Deployment deployment,
             return;
         }
 
+        if (newConfig.containsKey(SERVICES_NAMESPACE_TOPIC)) {
+            Map<String, Object> serviceConfig = (Map<String, Object>) newConfig.get(SERVICES_NAMESPACE_TOPIC);
+            if (serviceConfig.containsKey(deviceConfiguration.getNucleusComponentName())) {
+                Map<String, Object> kernelConfig =
+                        (Map<String, Object>) serviceConfig.get(DEFAULT_NUCLEUS_COMPONENT_NAME);
+                String awsRegion = tryGetAwsRegionFromNewConfig(kernelConfig);
+                String iotCredEndpoint = tryGetIoTCredEndpointFromNewConfig(kernelConfig);
+                String iotDataEndpoint = tryGetIoTDataEndpointFromNewConfig(kernelConfig);
+                try {
+                    deviceConfiguration.validateEndpoints(awsRegion, iotCredEndpoint, iotDataEndpoint);
+                } catch (DeviceConfigurationException e) {
+                    logger.atError().cause(e).log("Error validating IoT endpoints");
+                    return;
+                }
+            }
+        }
```

Comment:
Why do we need this code duplicated in both activator subclasses? we could extract it out to a common method in `DeploymentActivator.java`. Actually, I'd prefer the code to not live in the activators at all, but organize it alongside the rest of config validation here like this - https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/main/src/main/java/com/aws/greengrass/deployment/DeploymentConfigMerger.java#L122 
Also, can we complete the deployment future with `ComponentConfigurationValidationException` for consistency with existing validation? It helps convey the message more effectively to the customer